proxy/src/dns.rs

```diff
@@ -121,11 +121,14 @@ impl Resolver {
 
     // `ResolverFuture` can only be used for one lookup, so we have to clone all
     // the state during each resolution.
-    fn lookup_ip(self, &Name(ref name): &Name) -> LookupIpFuture {
+    fn lookup_ip(self, &Name(ref name): &Name)
+        // TODO: unbox me
+        -> Box<Future<Item = LookupIp, Error = ResolveError>> {
+        let name = name.clone();
```

Comment:
We have to clone this so the future is `'static` and can be `Box`ed. I'd like to eliminate the `String` cloning (at least by making `Names` reference counted, if not by eliminating the need to clone here...).